### DISCUSSION TITLE
How to: TwoWay CompiledBinding in code behind

### DISCUSSION THREAD

**Discussion by vpenades** [AvaloniaUI/Avalonia#17678] [General]
I am dynamically creating a context menu that binds a CheckBox.IsChecked to a Visual.IsVisible

The first time CheckBox is initialized, it takes the value of the visual, afterwards, the CheckBox updates the IsVisible property of the visual.

My current approach is:

```c#
var source = visual.GetObservable(Visual.IsVisibleProperty);
var binding = source.ToBinding();
checkbox.Bind(ToggleButton.IsCheckedProperty, binding);
```

The binding works.... in one direction;  the checkbox successfully gets the value from the IsVisible property... but it's unable to change it back.

Another detail worth to mention is that  CheckBox.IsChecked is of type nullable `bool?` whereas Visual.IsVisible is of type `bool`  , but certainly it's not affecting the current one-way binding.

So how can I set this binding as a two way?   Maybe using CompiledBindingExtension instead?

**Comment by StefanKoell** [comment]
Are you creating bindings between a VM and a control or between two controls?

But I guess you can just do the same the other way around? Create an observable/binding for the IsCheckedProperty of the checkbox and bind it to the visual?

**Reply by vpenades** [reply]
between two controls.

Also tried to create two bindings, each in the opposite direction. At first it seems it works, but the sencond time it begins behaving in weird way.

In any case, I think there's a lack of documentation in here:

[Avalonia Docs Compiled-Bindings](https://docs.avaloniaui.net/docs/basics/data/data-binding/compiled-bindings)

And they could also include information on how to do the same in code behind.

**Reply by StefanKoell** [reply]
Yeah, I've seen weird behavior myself when working with two controls. What I do in most cases, I override the OnPropertyChanged to update another control when a property value changed. You have more control in code and do conditional updates and prevent side effects or loops when you update another property. It depends on the use case, of course, but maybe this helps in your case as well.
